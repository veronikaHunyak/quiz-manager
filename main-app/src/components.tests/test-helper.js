export const mockQuiz = [
  {
    _id: "5eb14db7c8248d0f7cd4a279",
    quizKey: "quiz0",
    quizName: "First quiz test name",
    quizData: [
      {
        question: "Question 1",
        options: [{ A: "one" }, { B: "two" }, { C: "three" }],
        answer: "B",
      },
      {
        question: "question2",
        options: [{ A: "a" }, { B: "b" }, { C: "c" }, { D: "d" }],
        answer: "C",
      },
    ],
  },
];
