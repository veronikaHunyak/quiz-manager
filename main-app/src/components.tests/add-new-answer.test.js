import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import NewAnswer from "../components/add-new-answer";
import { mockQuiz } from "./test-helper";

describe("NewAnswer", () => {
  const props = {
    questionIndex: 0,
    quiz: mockQuiz[0],
  };

  it("should render correctly", () => {
    expect(toJson(shallow(<NewAnswer {...props} />))).toMatchSnapshot();
  });
});
