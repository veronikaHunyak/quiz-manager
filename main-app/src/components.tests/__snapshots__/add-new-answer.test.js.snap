// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAnswer should render correctly 1`] = `
<div>
  <div
    class="row"
    key="answer0"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      class="col-2"
    >
      <label
        for="option"
      >
        A
        :
      </label>
    </div>
    <div
      class="col-10"
    >
      <input
        class="form-control"
        id="option0"
        onChange={[Function]}
        placeholder="Enter an answer option"
        type="text"
      />
    </div>
  </div>
  <div
    class="row"
    key="answer1"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      class="col-2"
    >
      <label
        for="option"
      >
        B
        :
      </label>
    </div>
    <div
      class="col-10"
    >
      <input
        class="form-control"
        id="option1"
        onChange={[Function]}
        placeholder="Enter an answer option"
        type="text"
      />
    </div>
  </div>
  <div
    class="row"
    key="answer2"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      class="col-2"
    >
      <label
        for="option"
      >
        C
        :
      </label>
    </div>
    <div
      class="col-10"
    >
      <input
        class="form-control"
        id="option2"
        onChange={[Function]}
        placeholder="Enter an answer option"
        type="text"
      />
    </div>
  </div>
  <div
    class="row"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      class="col-2"
    >
      <label
        for="option"
      >
        Answer:
      </label>
    </div>
    <div
      class="col-10"
    >
      <div
        class="btn-group btn-group-toggle"
      >
        <label
          class="btn btn-secondary false"
          key="0"
        >
          <input
            autoComplete="off"
            id="options"
            name="options"
            onChange={[Function]}
            type="radio"
            value={
              Array [
                "A",
              ]
            }
          />
          A
        </label>
        <label
          class="btn btn-secondary active"
          key="1"
        >
          <input
            autoComplete="off"
            id="options"
            name="options"
            onChange={[Function]}
            type="radio"
            value={
              Array [
                "B",
              ]
            }
          />
          B
        </label>
        <label
          class="btn btn-secondary false"
          key="2"
        >
          <input
            autoComplete="off"
            id="options"
            name="options"
            onChange={[Function]}
            type="radio"
            value={
              Array [
                "C",
              ]
            }
          />
          C
        </label>
      </div>
    </div>
  </div>
</div>
`;
